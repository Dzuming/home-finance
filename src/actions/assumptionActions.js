import * as types from './actionTypes';
import { createAction } from 'redux-actions';
import env from '../../environments/config';

export const fetchAssumptions = period => (dispatch, getState) => {
  const { id } = getState().user;
  return dispatch({
    type: types.API_REQUEST_GET,
    payload: {
      url: `${env.api_url}/api/assumptions/${id}/${period}`,
      success: setAssumption,
    },
  });
};

export const fetchOverallAssumptions = () => (dispatch, getState) => {
  const { id } = getState().user;
  return dispatch({
    type: types.API_REQUEST_GET,
    payload: {
      url: `${env.api_url}/api/assumptions/${id}`,
      success: setOverallAssumptions,
    },
  });
};

export const fetchAssumptionTypes = period => (dispatch, getState) => {
  const { id } = getState().user;
  return dispatch({
    type: types.API_REQUEST_GET,
    payload: {
      url: `${env.api_url}/api/assumptionTypes/${id}/${period}`,
      success: setAssumptionTypes,
    },
  });
};

export const setAssumption = createAction(types.ASSUMPTION_CRUD.READ);

export const setOverallAssumptions = createAction(
  types.OVERALL_ASSUMPTIONS_CRUD.READ,
);

export const setAssumptionTypes = createAction(
  types.ASSUMPTION_TYPES_DRAG.READ,
);

export const createAssumption = createAction(types.API_REQUEST_POST, data => ({
  url: `${env.api_url}/api/assumptions`,
  success: addAssumption,
  data,
}));

export const addAssumption = createAction(
  types.ASSUMPTION_CRUD.CREATE,
  assumption => ({
    assumption,
  }),
);

export const reduceAssumptionTypes = createAction(
  types.ASSUMPTION_TYPES_DRAG.REDUCE,
  type => ({
    type,
  }),
);

export const reduceCategories = createAction(
  types.CATEGORIES_DRAG.REDUCE,
  type => ({
    type,
  }),
);
