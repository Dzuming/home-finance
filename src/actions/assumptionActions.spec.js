import * as actions from './assumptionActions';
import expect from 'expect';

const assumptions = [
  {
    id: 1,
    name: 'Poduszka bezpieczeństwa',
    percentage: 20,
    value: 66.6,
  },
  {
    id: 2,
    name: 'Wakacje',
    percentage: 10,
    value: 33.3,
  },
  {
    id: 3,
    name: 'Wakacje',
    percentage: 50,
    value: 695,
  },
];

const overallAssumptions = [
  {
    name: 'Poduszka bezpieczeństwa',
    value: 66.6,
  },
  {
    name: 'Wakacje',
    value: 33.3,
  },
  {
    name: 'Wakacje',
    value: 695,
  },
];

describe('assumption actions', () => {
  it('should create setAssumption action with snapshot', () => {
    expect(actions.setAssumption(assumptions)).toMatchSnapshot();
  });
  it('should create setOverallAssumptiona action with snapshot', () => {
    expect(actions.setOverallAssumptions(overallAssumptions)).toMatchSnapshot();
  });
  it('should create addAssumption action with snapshot', () => {
    const assumption = {
      userId: 1,
      assumptionTypeId: 1,
      percentage: 20,
      isInitialValue: 0,
      period: '2018-04',
    };
    expect(actions.addAssumption(assumption)).toMatchSnapshot();
  });
});
