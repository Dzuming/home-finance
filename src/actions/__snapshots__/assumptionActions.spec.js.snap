// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assumption actions should create addAssumption action with snapshot 1`] = `
Object {
  "payload": Object {
    "assumption": Object {
      "assumptionTypeId": 1,
      "isInitialValue": 0,
      "percentage": 20,
      "period": "2018-04",
      "userId": 1,
    },
  },
  "type": "CREATE_ASSUMPTION",
}
`;

exports[`assumption actions should create setAssumption action with snapshot 1`] = `
Object {
  "payload": Array [
    Object {
      "id": 1,
      "name": "Poduszka bezpieczeństwa",
      "percentage": 20,
      "value": 66.6,
    },
    Object {
      "id": 2,
      "name": "Wakacje",
      "percentage": 10,
      "value": 33.3,
    },
    Object {
      "id": 3,
      "name": "Wakacje",
      "percentage": 50,
      "value": 695,
    },
  ],
  "type": "READ_ASSUMPTION",
}
`;

exports[`assumption actions should create setOverallAssumptiona action with snapshot 1`] = `
Object {
  "payload": Array [
    Object {
      "name": "Poduszka bezpieczeństwa",
      "value": 66.6,
    },
    Object {
      "name": "Wakacje",
      "value": 33.3,
    },
    Object {
      "name": "Wakacje",
      "value": 695,
    },
  ],
  "type": "READ_OVERALL_ASSUMPTIONS",
}
`;

exports[`drag actions it should create resetDraggedAssumptionType action with a snapshot 1`] = `
Object {
  "type": "RESET_ASSUMPTION_TYPES",
}
`;

exports[`drag actions it should create selectAssumptionType action with a snapshot 1`] = `
Object {
  "payload": Object {
    "type": undefined,
  },
  "type": "REDUCE_ASSUMPTION_TYPES",
}
`;
